Common subdirectories: xhtml-3000.2.1/Text and xhtml-3000.2.2/Text
diff -u xhtml-3000.2.1/xhtml.cabal xhtml-3000.2.2/xhtml.cabal
--- xhtml-3000.2.1/xhtml.cabal	2014-08-20 02:52:46.686235983 -0400
+++ xhtml-3000.2.2/xhtml.cabal	2014-08-20 02:53:14.194372383 -0400
@@ -1,5 +1,5 @@
 Name:               xhtml
-Version:            3000.2.1
+Version:            3000.2.2
 Copyright:          Bjorn Bringert 2004-2006, Andy Gill and the Oregon
                     Graduate Institute of Science and Technology, 1999-2001
 Maintainer:         Chris Dornan <chris@chrisdornan.com>
